services/cdn/distributions.go

```diff
@@ -24,9 +24,15 @@ func SetDistributionData(d *schema.ResourceData, distribution cdn.Distribution)
 		}
 	}
 
+	// The Certificate ID must be read all the time, even if it is 'nil'. If we don't read it when
+	// it's 'nil', there will be some edge cases where the behavior will be incorrect.
 	if distribution.Properties.CertificateId != nil {
```

Comment:
Which edge cases?

Reply:
@cristiGuranIonos I modified the comment and included a full explanation, not sure how to make it shorter since it's a tricky scenario. I wanted to avoid this in the first place because it's a long comment but it doesn't help if it's not clear.